cli/flox/src/commands/environment.rs

```diff
@@ -1206,6 +1208,32 @@ impl Install {
                     {suggestion}
                 "})
             },
+            // If the package doesn't exist for this system, direct users to the docs
+            EnvironmentError2::Core(CoreEnvironmentError::LockedManifest(
+                LockedManifestError::BuildEnv(
+                    flox_rust_sdk::models::pkgdb::CallPkgDbError::PkgDbError(pkgdberr),
+                ),
+            )) if pkgdberr.exit_code == error_codes::PACKAGE_EVAL_INCOMPATIBLE_SYSTEM => {
+                debug!("incompatible package, directing user to docs");
+                let msg = if let Some(ctx_msg) = pkgdberr.context_message {
+                    formatdoc! {"
```

Comment:
note: this duplicates some logic in errors.rs, and the pattern here is different than what we have in `Pull`:
```
                let pkgdb_error = format_locked_manifest_error(builder_error);
```
I personally would prefer the logic lived on `PkgDbError`, and both `Pull` and `Install` could just do `format!({pkgdberr} ...)`

Reply:
The intention of `errors.rs` is to provide _additional_ context for errors in a categoric command independent way.

I think this particular error could as well be handled categorically..

We can move more of errors.rs into `Display` impls of the individual errors, as long as we keep adding application context (hints to specific commands, env variables, etc.) separately and move away from the dense error chain style of errors (or support both explicitly using [alternate](https://doc.rust-lang.org/beta/std/fmt/struct.Formatter.html#method.alternate)).

Reply:
But don't we need to catch it here to add the link? The way I read https://github.com/flox/flox/issues/864 we don't want to add the think for pull until a choice is made by the user